be/src/formats/json/map_column.cpp

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "column/map_column.h"
+
+#include "fmt/format.h"
+#include "formats/json/map_column.h"
+#include "formats/json/nullable_column.h"
+#include "gutil/strings/substitute.h"
+
+namespace starrocks {
+
+Status add_map_column(Column* column, const TypeDescriptor& type_desc, const std::string& name,
+                      simdjson::ondemand::value* value) {
+    auto map_column = down_cast<MapColumn*>(column);
+
+    try {
+        simdjson::ondemand::object obj = value->get_object();
```

Comment:
if value is not an object, what will be the error message?

Reply:
An exception `The JSON element does not have the requested type.` will be thrown and be caught later.

Reply:
Can we give more detailed info, such as the expected type is xxx, the real given type is yyy. It will be much clearer.